install.sh

```diff
@@ -33,18 +33,18 @@ fi
 echo "Gateway configuration:"
 
 # Try to get gateway ID from MAC address
-# First try eth0, if that does not exist, try wlan0 (for RPi Zero)
-GATEWAY_EUI_NIC="eth0"
-if [[ `grep "$GATEWAY_EUI_NIC" /proc/net/dev` == "" ]]; then
-    GATEWAY_EUI_NIC="wlan0"
-fi
 
-if [[ `grep "$GATEWAY_EUI_NIC" /proc/net/dev` == "" ]]; then
+# Get first non-loopback network device that is currently connected
+NICS=$(ip link 2>&1 | grep "state UP" | grep -v LOOPBACK | sed -E 's/^[0-9]+: ([0-9a-z]+): .*/\1/')
+if [[ -z $NICS ]]; then
     echo "ERROR: No network interface found. Cannot set gateway ID."
     exit 1
 fi
+GATEWAY_EUI_NIC=$(echo $NICS | cut -d " " -f1)
 
-GATEWAY_EUI=$(ip link show $GATEWAY_EUI_NIC | awk '/ether/ {print $2}' | awk -F\: '{print $1$2$3"FFFE"$4$5$6}')
```

Comment:
Are you injecting the `FFFE`part in between as well? on a quick glance I don't see it, and that would generate a shorter EUI (48 bits instead of 64 iirc)

Reply:
Ah, thanks for noticing!